test/metabase/db/query_test.clj

```diff
@@ -76,24 +76,22 @@
                :database (mt/id)}]
         (verify-same-query q)))))
 
-(deftest nonsql-dialects-return-original-string-test
-  (testing "Passing a mongodb query through format-sql returns the original string"
-    (with-open [r (io/reader (io/resource "metabase/db/mongodbquery.json"))]
-      (let [query                 (slurp r)
-            ;; Formatting a non-sql string returns nothing
-            formatted-query       (mdb.query/format-sql query :mongo)
-            ;; This is a mongodb query, but if you pass in the wrong driver it will attempt the format
-            ;; This is a corner case since the system should always be using the right driver
-            weird-formatted-query (mdb.query/format-sql query :postgres)]
-        (testing "Formatting a non-sql query returns the same string"
-          (is (= query formatted-query)))
-        (testing "The wrong formatter will change the format..."
-          (is (not= query weird-formatted-query)))
-        (testing "...but the resulting data is still the same"
-          ;; Bottom line - Use the right driver, but if you use the wrong
-          ;; one it should be harmless but annoying
-          (is (= (json/parse-string query)
-                 (json/parse-string weird-formatted-query))))))))
```

Comment:
Why did you remove these tests?

Reply:
The primary task for this test suite was to ensure we didn't get weird formatting as in `format-sql-with-params-test` below the above (See the expansion of { { } }). I updated the formatter for this. The mongo tests were there to capture behavior for if you happen to use the format-sql function on a mongo query string. I haven't looked at your changes, but that's the context.

Reply:
yea but why do we need to handle that case? In which case we could pass the wrong driver? 

I can put that test back but I don't think allowing vagueness is a good practice.

Reply:
Saw Mark is OOO and I want to lend this so I revert back those tests b3a96f29c3